ksql-engine/src/main/java/io/confluent/ksql/function/UdfLoader.java

```diff
@@ -133,11 +157,31 @@ private void addFunction(final UdfDescription annotation,
         }));
   }
 
+  private void addSensor(final String sensorName, final String udfName) {
```

Comment:
We also need a way to remove the sensor from the metrics registry when the query terminates. Otherwise we will have a metrics leak. The problem will be especially acute if a transient query is run and the global config has `ksql.udf.collect.metrics` == `true`. the metrics for every UDF in transient queries will always linger.

Reply:
@apurvam as it is at the moment these are global metrics. So there is no leak. Not sure if they need to be per query as it is really for debugging, i.e., i wouldn't recommend having it set to true for production usage.